File: 2nd_question_the_corridor_or_there_and_back_again.java
import java.util.Scanner;
public class Main{
    public static void main(String[] args){
        Scanner sc = new Scanner(System.in);
        int t = sc.nextInt();
        while(t-- > 0){
            int number_of_traps = sc.nextInt();
            //ans stores the maximum possible value of k
            int ans = Integer.MAX_VALUE;
            for (int i = 0; i < number_of_traps; i++){
                int room_number = sc.nextInt();
                int time_limit = sc.nextInt();            
                //As we are looking for the ceil value, we need to add 1 to time_limit and then divide the whole by 2.
                ans = Math.min(ans, (room_number + (time_limit + 1) / 2 - 1));
            }
            System.out.println(ans);
        }
        sc.close();
    }
}

File: 2nd_question_the_corridor_or_there_and_back_again.cpp
#include<bits/stdc++.h>
using namespace std;
 
int main(){
    int t;
    cin >> t;
    while(t--){
        int number_of_traps;
        cin >> number_of_traps;
        int ans = INT_MAX;
        for(int i = 0; i < number_of_traps; i++){
            int room_number, time_limit;
            cin >> room_number >> time_limit;
            //ans stores the maximum possible value of k
            //As we are looking for the ceil value, we need to add 1 to time_limit and then divide the whole by 2.
            ans = min(ans, (room_number + (time_limit + 1)/2 - 1));
        }
        cout << ans << endl;
    }
    return 0;
}
